demos/data/scripts/generator/PersonGen.py

```diff
@@ -0,0 +1,438 @@
+                "state": address.get("state", "Unknown State"),
+                "zip": address.get("postalCode", "Unknown PostalCode"),
+                "lat": address.get("coordinates", {}).get("lat", 0.0),
+                "lon": address.get("coordinates", {}).get("lng", 0.0)
+            }
+            records.append(record)
+        df = pd.DataFrame(records)
+        return df
+
+    def generate_call_logs(
+            self,
+            people_df: pd.DataFrame,
+            num_logs: int = 500,
+            start_date: str = '-1y'
+            ) -> pd.DataFrame:
+        """
+        Generate call logs for a set of people, simulating everyday calls.
+        :param people_df: DataFrame of people with affiliations.
+        :param num_logs: Number of call logs to generate.
+        :param start_date: Start date for call logs.
+        :return: DataFrame of call logs.
+        """
+        call_logs = []
+        phone_numbers = people_df['phone_number'].tolist()
+        
+        for _ in range(num_logs):
+            caller, callee = self.random.sample(phone_numbers, 2)  # Ensure caller and callee are different
+            call_date = self.fake.date_time_between(start_date=start_date)
+            call_time = call_date + timedelta(hours=self.random.randint(0, 23), minutes=self.random.randint(0, 59), seconds=self.random.randint(0, 59))
+            duration = self.random.randint(1, 3600)  # Call duration in seconds, from 1 sec to 1 hour
+            
+            call_logs.append({
+                "caller": caller,
+                "callee": callee,
+                "call_date": call_date.strftime('%Y-%m-%d'),
+                "call_time": call_time.strftime('%H:%M:%S'),
+                "duration_sec": duration
+            })
+        
+        return pd.DataFrame(call_logs)
+
+    def generate_non_affiliated_call_logs(
+            self,
+            people_df: pd.DataFrame,
+            call_logs_df: pd.DataFrame,
+            num_calls: int = 500,
+            start_date: str = '-1y'
+            ) -> pd.DataFrame:
+        """
+        Generate call logs for non-affiliated individuals, simulating everyday calls.
+
+        :param people_df: DataFrame of people with affiliations.
+        :param call_logs_df: DataFrame of call logs to append to.
+        :param num_calls: Number of calls to generate among non-affiliated individuals.
+        :return: Updated DataFrame with non-affiliated call logs.
+        """
+        # Filter for non-affiliated individuals
+        non_affiliated_people = people_df[people_df['affiliation'] == 'None']
+        
+        # Generate call logs
+        for _ in range(num_calls):
+            if len(non_affiliated_people) > 1:
+                caller, callee = non_affiliated_people.sample(n=2, replace=False)['phone_number'].values
+                self.add_call_log(call_logs_df, caller, callee, start_date)
+
+        return call_logs_df
+
+    def generate_affiliated_call_logs(
```

Comment:
is this modeling phenomena like burner phones vs regular? i'm not sure how that'd look

Reply:
Oh no, this is gang affiliated call logs versus regular, burner would be interesting, not sure how to model that though , maybe like similar profiles like someone calls x,y,z numbers on main, but then realized he slipped up and switches to a burner and calls same numbers?

Reply:
yeah i think you'd want the distributions to mirror

so like your simulation secretly tracks who has what burner when, and has burners call one another most of the time, and occasionally the slipup of calling a main